python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -143,3 +143,180 @@ def columns(self):
+
+    The color difference (color1 - color2) is then scaled by the error on
+    the color as computed from color1_flux1_err, color1_flux2_err,
+    color2_flux1_err, color2_flux2_err.  The errors on color2 may be omitted
+    if the comparison is between an "observed" catalog and a "truth" catalog.
+
+    Parameters
+    ----------
+    df : `pandas.core.frame.DataFrame`
+        The catalog to calculate the color difference from.
+
+    Returns
+    -------
+    The color difference scaled by the error.
+
+    Notes
+    -----
+    The flux columns need to be in units that can be converted
+    to janskies. This action doesn't have any calibration
+    information and assumes that the fluxes are already
+    calibrated.
+    """
+    color1_flux1_err = Field(doc="Error column for flux1 for color1",
+                             dtype=str)
+    color1_flux2_err = Field(doc="Error column for flux2 for color1",
+                             dtype=str)
+    color2_flux1_err = Field(doc="Error column for flux1 for color2",
+                             dtype=str,
+                             default="")
+    color2_flux2_err = Field(doc="Error column for flux2 for color2",
+                             dtype=str,
+                             default="")
+
+    @property
+    def columns(self):
+        cols = (self.color1_flux1,
+                self.color1_flux2,
+                self.color2_flux1,
+                self.color2_flux2,
+                self.color1_flux1_err,
+                self.color1_flux2_err)
+
+        if self.color2_flux1_err and self.color2_flux2_err:
+            cols = cols + (self.color2_flux1_err,
+                           self.color2_flux2_err)
+
+        return cols
+
+    def __call__(self, df):
+        k = 2.5/np.log(10.)
+
+        color1_flux1 = df[self.color1_flux1].values*u.Unit(self.color1_flux1_units)
+        color1_mag1 = color1_flux1.to(u.ABmag).value
+        color1_mag1_err = k*df[self.color1_flux1_err].values/df[self.color1_flux1].values
+
+        color1_flux2 = df[self.color1_flux2].values*u.Unit(self.color1_flux2_units)
+        color1_mag2 = color1_flux2.to(u.ABmag).value
+        color1_mag2_err = k*df[self.color1_flux2_err].values/df[self.color1_flux2].values
+
+        color2_flux1 = df[self.color2_flux1].values*u.Unit(self.color2_flux1_units)
+        color2_mag1 = color2_flux1.to(u.ABmag).value
+        if self.color2_flux1_err:
+            color2_mag1_err = k*df[self.color2_flux1_err].values/df[self.color2_flux1].values
+        else:
+            color2_mag1_err = 0.0
+
+        color2_flux2 = df[self.color2_flux2].values*u.Unit(self.color2_flux2_units)
+        color2_mag2 = color2_flux2.to(u.ABmag).value
+        if self.color2_flux2_err:
+            color2_mag2_err = k*df[self.color2_flux2_err].values/df[self.color2_flux2].values
+        else:
+            color2_mag2_err = 0.0
+
+        color1 = color1_mag1 - color1_mag2
+        err1_sq = color1_mag1_err**2. + color1_mag2_err**2.
```

Comment:
You could consider `np.hypot` of all for columns instead of squaring and taking `sqrt` later.

Reply:
Upon further review, `np.hypot` only works on two arrays/values, so it's back to square (root) one.

Reply:
Fair enough. All looks good to me.